File: node-app/tictactoe/game.js

function game(){
    var obj={};

    obj.players=[];
	obj.analyzer;
	obj.board;
	var currentPlayer=0;
	obj.createGame=function(players,analyzer,board ){
		obj.analyzer=analyzer;
		obj.board=board;
		obj.players=players;
	}
	
	obj.play=function(cellno){
			currentPlayer++;
			if(currentPlayer%2==0){
				obj.board.cellSet(cellno,obj.players[1].mark);
			}
			else{
				obj.board.cellSet(cellno,obj.players[0].mark);
			}
	}

    return obj;
}
module.exports=game;

File: TicTacToe/app/src/main/java/com/techlabs/paresh/tic_tac_toe/Game.java
package com.techlabs.paresh.tic_tac_toe;

public class Game {
	Player players[];
	ResultAnalyzer analyzer;
	Board board;
	int currentPlayer=0;
	public Game(Player[] players,ResultAnalyzer analyser,Board board ){
		this.analyzer=analyser;
		this.board=board;
		this.players=players;
	}
	
	public void play(int cellno){
			currentPlayer++;
			if(currentPlayer%2==0){
				board.cellSet(cellno,players[1].mark);
			}
			else{
				board.cellSet(cellno,players[0].mark);
			}
	}
}
